src/binder/bind_node_visitor.cpp

```diff
@@ -454,6 +455,75 @@ void BindNodeVisitor::Visit(common::ManagedPointer<parser::SelectStatement> node
   BinderContext context(context_);
   context_ = common::ManagedPointer(&context);
 
+  for (auto &ref : node->GetSelectWith()) {
+    // Store CTE table name
+    cte_table_name_.push_back(ref->GetAlias());
+
+    bool inductive =
```

Comment:
Nit: short comment on what inductive means? Or a general high-level explanation of what this block of code does here.

Reply:
In general it is not obvious to me what the new code is doing and how to extend it if necessary, I think an overview comment would be very helpful here.